drivers/display/ssd16xx.c

```diff
@@ -106,12 +109,12 @@ static inline int ssd16xx_write_cmd(struct ssd16xx_data *driver,
 
 static inline void ssd16xx_busy_wait(struct ssd16xx_data *driver)
 {
-	u32_t val = 0U;
+	int pin = gpio_pin_get(driver->busy, SSD16XX_BUSY_PIN);
 
-	gpio_pin_read(driver->busy, SSD16XX_BUSY_PIN, &val);
-	while (val) {
+	while (pin) {
```

Comment:
This should be `pin > 0` to prevent an infinite loop if the GPIO driver can return an error on read (e.g. SX1509B with an I2C failure).  (Or `pin ==- 1` but I prefer "positive means set".)